lib/node/index.js

```diff
@@ -595,6 +598,8 @@ Request.prototype.request = function(){
     // because node will emit a faux-error "socket hang up"
     // when request is aborted before a connection is made
     if (self._aborted) return;
+    // flag retry here for out timeouts (same as above)
+    if (self._retries) return;
```

Comment:
I don't understand why it skips errors here. 

Also note that all these handlers are `.once()`, so they should have been re-attached again on retry. That's why I've mentioned earlier that .end() may need to be split up into once-ever and once-per-retry functions.